src/ProgressOnderwijsUtils/Collections/Tree.cs

```diff
@@ -128,12 +128,9 @@ public bool Equals(Tree<T>? a, Tree<T>? b)
 
             [Pure]
             bool ShallowEquals(NodePair pair)
-                // ReSharper disable RedundantCast

```

Comment:
sidenote: ik heb wel eens overwogen of we niet een keer al dit soort dingen (en pragma disables) een keer weg moeten regexen. Ik vermoed dat veel meer onterecht zijn inmiddels

Reply:
Sortgelijke issues zijn er ook met AssertNotNull en in mindere mate ook PretendNullable.

Het is wel een interessant probleem iig.